Library/Homebrew/cmd/list.rb

```diff
@@ -132,7 +132,9 @@ def list
       system_command! "find", args:         args.named.to_default_kegs.map(&:to_s) + %w[-not -type d -print],
                               print_stdout: true
     else
-      args.named.to_default_kegs.each { |keg| PrettyListing.new keg }
+      formula_names, cask_names = args.named.to_resolved_formulae_to_casks
+      formula_names.map(&:name).each { |keg| PrettyListing.new keg } if formula_names.present?
+      list_casks(cask_names, args) if cask_names.present?
```

Comment:
If `formula_names` is present but `cask_names` is not, won't nothing show? I think in that case we still want the formula names to show, right?